core/trino-main/src/main/java/io/trino/execution/TaskManagerConfig.java

```diff
@@ -447,6 +448,21 @@ public TaskManagerConfig setScaleWritersMaxWriterCount(int scaleWritersMaxWriter
         return this;
     }
 
+    @Min(1)
+    public int getScaleWritersMaxNodesCount()
+    {
+        return scaleWritersMaxNodesCount;
+    }
+
+    @Config("task.scale-writers.max-nodes-count")
```

Comment:
This should be moved to `QueryManagerConfig` below `query.hash-partition-count` setting.
Name it `query.max-writer-node-count` or something similar.